File: go-sdk/pkg/models/operations/postauthorize.go
package operations

import (
	"openapi/pkg/models/shared"
)

type PostAuthorizePathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type PostAuthorizeRequest struct {
	PathParams PostAuthorizePathParams
	Request    *shared.CreateAuthorization `request:"mediaType=application/json"`
}

type PostAuthorizeResponse struct {
	Authorization *shared.Authorization
	ContentType   string
	Error         *shared.Error
	StatusCode    int64
}

File: typescript-sdk/src/sdk/models/operations/postauthorize.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostAuthorizePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class PostAuthorizeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostAuthorizePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.CreateAuthorization;
}


export class PostAuthorizeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  authorization?: shared.Authorization;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;
}
